sdk/core/src/tasks/sboms/github/sync.rs

```diff
@@ -0,0 +1,409 @@
+
+        let mut errors = HashMap::new();
+
+        let mut repos: Vec<Repo> = match self.github.get_repos().await {
+            Ok(value) => value,
+            Err(err) => {
+                println!("Error attempting to get the repos: {}", err);
+                task.err_total += 1;
+                Vec::new()
+            }
+        };
+
+        let mut total = 0;
+        let mut failed = 0;
+        let mut skipped = 0;
+
+        for repo in &mut repos {
+
+            let name = repo.full_name.clone().unwrap_or(
+                String::from("name/missing")
+            );
+
+            let url = match repo.html_url.clone() {
+                Some(url) => url,
+                None => continue
+            };
+
+            if !SyncTask::should_skip(repo, name, url.clone()) {
+
+                match self.process_repo(repo, task).await {
+
+                    Ok(option) => match option {
+                        Some(raw) => {
+                            match self.sboms.ingest(
+                                    raw.as_str(),
+                                    None,
+                                    SbomProviderKind::GitHub,
+                                    Xref {
+                                        kind: XrefKind::Product,
+                                        map: HashMap::new()
+                                    },
+                                    task,
+                                )
+                                .await {
+                                Ok(sbom) => {
+                                    let opt = Some(sbom);
+                                    println!("==> SBOM Ingestion went OK");
+                                    opt
+                                },
+                                Err(err) => {
+                                    task.err_total += 1;
+                                    errors.insert(url.clone(), err.to_string());
+                                    println!("SBOM Ingestion FAILED");
+                                    failed += 1;
+                                    None
+                                }
+                            };
+                        }
+                        None => println!("==> No work to do because the hashes matched")
+                    }
+                    Err(err) => println!("==> Repo processing failure: {}", err)
+                };
+            } else {
+                skipped += 1;
+            }
+            total += 1;
+        }
+
+        // TODO How should I actually report these?
+        println!("==> FINISHED PROCESSING, total: {}, failed: {}, skipped: {}", total, failed, skipped);
+
+        Ok(errors)
+    }
+}
+
+impl Service<Task> for SyncTask {
+    fn store(&self) -> Arc<Store> {
+        self.mongo_svc.store.clone()
+    }
+}
+
+impl SyncTask {
+
+    /* PRIVATE */
+
+    /// Invokes the syft CLI against the cloned repository to generate an SBOM.
+    fn syft(
```

Comment:
This is completely fine for now, but like we discussed, I do want to move this to some sort of `syft` module in the future. The CLI is going to need this for example, without invoking a sync task.

Reply:
Nice yes, I'll at it top my personal backlog and suggest it at planning if it is appropriate